### PR TITLE
Add pre-commit hook

### PR DESCRIPTION
**Pull request by alexifm**
This creates a pre-commit hook for flakehell.

### CODE REVIEW

**Comment by alexifm** [comment]
Got it.  Thanks.

**Comment by mjpieters** [comment]
Another issue I see: flakehell lints more than just Python files, so `types: [python]` is too strict. Better to use `types: [python, markdown, rst, jupyter, yaml]`, to cover [all supported parsers](https://flakehell.readthedocs.io/parsers.html#parsers).

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-   id: flakehell
+    name: flakehell
+    description: '`flakehell` is `flake8` wrapper to make a nice, legacy-friendly, configurable command-line utility for enforcing style consistency across Python projects.'
```
Comment (mjpieters) [inline .pre-commit-hooks.yaml:3]: Grammar nitpick:

> `flakehell` is [**_a_**] `flake8 ` wrapper to make it nice, legacy-friendly, configurable ...

(without the square brackets, those are just there to make sure the single-letter word `a` stands out 😃 )

**Comment by alexifm** [comment]
Done.  I didn't see all the types/extensions in the documentation previously so I just went with python.

**Comment by mjpieters** [comment]
To the project maintainers: for what it's worth, I'm using this pull request in a project now:

```yaml
repos:
  - repo: https://github.com/alexifm/flakehell
    # this is a pull request for the official project, at
    # https://github.com/life4/flakehell/pull/85
    rev: 44458e7baf4081f2d9df64a7169ea20eb2a3231c
    hooks:
      - id: flakehell
        additional_dependencies:
          - flake8-bugbear
```

If this is merged, I can switch it to:

```yaml
repos:
  - repo: https://github.com/life4/flakehell
    rev: 0.5.1
    hooks:
      - id: flakehell
        additional_dependencies:
          - flake8-bugbear
```

The `rev` value will of course depend on what future tags include the `.pre-commit-hooks.yaml` file. Note the `additional_dependencies` entry there, that's where you'd list any extra plugins to install.

**Comment by orsinium** [comment]
Great! Thank you, that's beautiful :)

**Comment by and-semakin** [comment]
Unfortunately, it doesn't work for me. I use the following config:

```
  - repo: https://github.com/life4/flakehell
    rev: 56b2ac7f9a2895546ac742b7bfa70babe2730971
    hooks:
      - id: flakehell
        types: [python]
        additional_dependencies:
          - flake8-bugbear
          - flake8-pytestrail
          - flake8-docstrings
          - darglint
```

Notice, I had to specify `types: [python]`. Without it `pre-commit` skips checks with the following text:

```
flakehell............................................(no files to check)Skipped
```

That's the error I receive when try to commit changed Python file:

```
$ git commit -v
flakehell................................................................Failed
- hook id: flakehell
- exit code: 1

Executable `flakehell` not found
```

Let's try to figure out what's wrong in `~/.cache/pre-commit` directory. Here I have two other directories. One of them is just pure `flakehell` repo without any additional files:

```
$ pwd
/Users/andreisemakin/.cache/pre-commit/repo1lxznxps
$ ls -la
total 80
drwx------  16 andreisemakin  staff   512 Jul  7 15:05 ./
drwxr-xr-x  13 andreisemakin  staff   416 Jul  7 15:06 ../
-rw-r--r--   1 andreisemakin  staff   405 Jul  7 15:05 .editorconfig
drwxr-xr-x  10 andreisemakin  staff   320 Jul  7 15:05 .git/
-rw-r--r--   1 andreisemakin  staff    96 Jul  7 15:05 .gitignore
-rw-r--r--   1 andreisemakin  staff   336 Jul  7 15:05 .pre-commit-hooks.yaml
-rw-r--r--   1 andreisemakin  staff  1322 Jul  7 15:05 .travis.yml
-rw-r--r--   1 andreisemakin  staff  1080 Jul  7 15:05 LICENSE
-rw-r--r--   1 andreisemakin  staff  4841 Jul  7 15:05 README.md
drwxr-xr-x  14 andreisemakin  staff   448 Jul  7 15:05 assets/
drwxr-xr-x  11 andreisemakin  staff   352 Jul  7 15:05 docs/
-rw-r--r--   1 andreisemakin  staff   806 Jul  7 15:05 example.py
drwxr-xr-x  14 andreisemakin  staff   448 Jul  7 15:05 flakehell/
-rw-r--r--   1 andreisemakin  staff    98 Jul  7 15:05 markdownlint.json
-rw-r--r--   1 andreisemakin  staff  3887 Jul  7 15:05 pyproject.toml
drwxr-xr-x   8 andreisemakin  staff   256 Jul  7 15:05 tests/
```

Another one contains the same files but also contains new virtualenv `py_env-python3.8/`:

```
$ pwd
/Users/andreisemakin/.cache/pre-commit/repoy9z_9gxz
$ ls -la
total 80
drwx------  17 andreisemakin  staff   544 Jul  7 15:06 ./
drwxr-xr-x  13 andreisemakin  staff   416 Jul  7 15:06 ../
-rw-r--r--   1 andreisemakin  staff   405 Jul  7 15:06 .editorconfig
drwxr-xr-x  10 andreisemakin  staff   320 Jul  7 15:06 .git/
-rw-r--r--   1 andreisemakin  staff    96 Jul  7 15:06 .gitignore
-rw-r--r--   1 andreisemakin  staff   336 Jul  7 15:06 .pre-commit-hooks.yaml
-rw-r--r--   1 andreisemakin  staff  1322 Jul  7 15:06 .travis.yml
-rw-r--r--   1 andreisemakin  staff  1080 Jul  7 15:06 LICENSE
-rw-r--r--   1 andreisemakin  staff  4841 Jul  7 15:06 README.md
drwxr-xr-x  14 andreisemakin  staff   448 Jul  7 15:06 assets/
drwxr-xr-x  11 andreisemakin  staff   352 Jul  7 15:06 docs/
-rw-r--r--   1 andreisemakin  staff   806 Jul  7 15:06 example.py
drwxr-xr-x  14 andreisemakin  staff   448 Jul  7 15:06 flakehell/
-rw-r--r--   1 andreisemakin  staff    98 Jul  7 15:06 markdownlint.json
drwxr-xr-x   7 andreisemakin  staff   224 Jul  7 15:06 py_env-python3.8/
-rw-r--r--   1 andreisemakin  staff  3887 Jul  7 15:06 pyproject.toml
drwxr-xr-x   8 andreisemakin  staff   256 Jul  7 15:06 tests/
```

Let's see what's installed inside of this new virtualenv:

```
(venv) $ pip freeze
attrs==19.3.0
darglint==1.5.1
flake8==3.8.3
flake8-bugbear==20.1.4
flake8-docstrings==1.5.0
flake8-pytestrail==0.2.0
mccabe==0.6.1
pycodestyle==2.6.0
pydocstyle==5.0.2
pyflakes==2.2.0
snowballstemmer==2.0.0
UNKNOWN @ file:///Users/andreisemakin/.cache/pre-commit/repoy9z_9gxz
```

As you can see, there is no `flakehell`. I guess that's why it can't find it.

Ok, let's try to specify `flakehell` as a dependency of `flakehell` (dirty hack, but let's give it a try):

```
  - repo: https://github.com/life4/flakehell
    rev: 56b2ac7f9a2895546ac742b7bfa70babe2730971
    hooks:
      - id: flakehell
        types: [python]
        additional_dependencies:
          - flakehell  # <-- here
          - flake8-bugbear
          - flake8-pytestrail
          - flake8-docstrings
          - darglint
```

It works this way:

```
flakehell................................................................Failed
- hook id: flakehell
- exit code: 1

project/connector/__init__.py
     1:   1 D400 First line should end with a period
```

Any ideas on what could have gone wrong? Could it be somehow connected to the fact that `flakehell` is a `pyproject.toml` package?

**Comment by mjpieters** [comment]
> Notice, I had to specify types: [python]. 

I rechecked, and you are correct; I misstood how the `types` list is matched. It is specifies the minimal subset of tags each file must have (so take the output of `identify-cli  filename` and only files that have _all_ the tags named in the `types` list are matched. I'll submit a fix; the correct config would be:

```yaml
        types: [file]
        files: \.(ipynb|md|py|rst|yaml|yml)$
```

> That's the error I receive when try to commit changed Python file:
> 
>     Executable `flakehell` not found

That's https://github.com/life4/flakehell/pull/88; flakehell can't actually be installed from source as it is missing a crucial PEP 518 section.

It happened to work for me because I _also_ had flakehell installed in the parent Python environment, and so was found on in another location on the `PATH` search path.

**Comment by mjpieters** [comment]
I've submitted #89 to fix the `types` issue; sorry about that.

**Pull request merged**